packages/evolution-common/src/services/baseObjects/__tests__/Address.test.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2024, Polytechnique Montreal and contributors
+ *
+ * This file is licensed under the MIT License.
+ * License text available at https://opensource.org/licenses/MIT
+ */
+
+import { Address, AddressAttributes, ExtendedAddressAttributes, addressAttributes } from '../Address';
+import { v4 as uuidV4 } from 'uuid';
+import { isOk, hasErrors, unwrap } from '../../../types/Result.type';
+
+describe('Address', () => {
+    const validAddressAttributes: AddressAttributes = {
+        _uuid: uuidV4(),
+        _isValid: true,
+        civicNumber: 123,
+        civicNumberSuffix: 'A',
+        unitNumber: 456,
+        streetName: 'Main Street',
+        streetNameHomogenized: 'main street',
+        streetNameId: 'street-id',
+        streetNameInternalId: 'street-internal-id',
+        municipalityName: 'City',
+        municipalityCode: 'CITY',
+        postalMunicipalityName: 'Postal City',
+        region: 'Region',
+        country: 'Country',
+        postalCode: 'A1B 2C3',
+        addressId: 'address-id',
+        internalId: 'internal-id',
+    };
+
+    const extendedAddressAttributes: ExtendedAddressAttributes = {
+        ...validAddressAttributes,
+        customAttribute: 'custom value',
+    };
+
+    test('should have a validateParams section for each attribute', () => {
+        const validateParamsCode = Address.validateParams.toString();
+        addressAttributes.forEach((attributeName) => {
+            expect(validateParamsCode).toContain(attributeName);
+        });
+    });
+
+    test('should get uuid', () => {
+        const address = new Address({ ...validAddressAttributes, _uuid: '11b78eb3-a5d8-484d-805d-1f947160bb9e' });
+        expect(address._uuid).toBe('11b78eb3-a5d8-484d-805d-1f947160bb9e');
+    });
+
+    test('should create an Address instance with valid attributes using constructor', () => {
+        const address = new Address(validAddressAttributes);
+        expect(address).toBeInstanceOf(Address);
+        expect(address.attributes).toEqual(validAddressAttributes);
+    });
+
+    test('should return an error for invalid params', () => {
+        const invalidAttributes = 'foo' as any;
+        const result = Address.create(invalidAttributes);
+        expect(hasErrors(result)).toBe(true);
+        expect((unwrap(result) as Error[])).toHaveLength(1);
+    });
+
+    test('should create an Address instance with valid attributes', () => {
+        const result = Address.create(validAddressAttributes);
+        expect(isOk(result)).toBe(true);
+        expect(unwrap(result)).toBeInstanceOf(Address);
+    });
+
+    test('should create an Address instance with extended attributes', () => {
+        const result = Address.create(extendedAddressAttributes);
```

Comment:
Tu peux probablement merger ce test avec le test intitulé "'should create an Address instance with custom attributes'" plus bas, c'est le même

Reply:
removed